src/Datadog.Trace.ClrProfiler.Managed/Integrations/ServiceStackRedisIntegration.cs

```diff
@@ -42,12 +46,23 @@ public static class ServiceStackRedisIntegration
             int opCode,
             int mdToken)
         {
-            var originalMethod = Emit.DynamicMethodBuilder<Func<object, byte[][], object, object, bool, T>>
-               .GetOrCreateMethodCallDelegate(
-                    redisNativeClient.GetType(),
-                    "SendReceive",
-                    (OpCodeValue)opCode,
-                    methodGenericArguments: new[] { typeof(T) });
+            Func<object, byte[][], object, object, bool, T> instrumentedMethod = null;
+
+            try
+            {
+                var instrumentedType = redisNativeClient.GetInstrumentedType(RedisNativeClient);
```

Comment:
Any reason we don't just use `Type.GetType()`?

Reply:
My thinking was we're assured that the type is present in instance situations and it's a safer, potentially faster way to resolve.

Reply:
According to the static `Type.GetType` method docs: 
>>> The assembly-qualified name of the type to get. See AssemblyQualifiedName. If the type is in the currently executing assembly or in Mscorlib.dll, it is sufficient to supply the type name qualified by its namespace.

Reply:
> The lookup wouldn't have to scan each assembly.

The `string typeName` parameter is an assembly-qualified type name, so it won't scan each assembly.

> If it's not found it throws a `TypeLoadException`

There is a [`Type.GetType(string typeName, bool throwOnError)`](https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype#System_Type_GetType_System_String_System_Boolean_) overload that returns `null` instead of throwing an exception if we pass `false` as the second argument.

Reply:
If we're instrumenting an instance method, do we not already have the type on that hierarchy?
`static Type.GetType(string name)` looks at executing assembly and `mscorlib` doesn't it?
It actually feels safer and more straight-forward to me to retrieve it from the instance when we have that option.

Reply:
What I'm suggesting is:
```suggestion
                var instrumentedType = Type.GetType("ServiceStack.Redis.RedisNativeClient, ServiceStack.Redis");
```
We are providing the assembly name, so that's where it will look. We know the type and assembly at compile time, so this type can be retrieved once and cached into a static readonly field if we wanted to.

Reply:
I'm okay with using either one, IMO the instance.GetType is simpler/safer, and I'm not sure about caching the type either, seems like an entry point for a bug.